### PR TITLE
calculator and secret number homework

### PR DESCRIPTION
**Pull request by s-taylor**

### CODE REVIEW

File: students/simon_taylor/secret_number.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
+def secret_number
+  number = random_num
```
Comment (mathildathompson) [inline students/simon_taylor/secret_number.rb:2]: Nice use of calling a method within another method!

File: students/simon_taylor/secret_number.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      end
+
+    elsif guess > number
+      puts "Sorry but your guess was too high, please try again"
+    elsif guess < number
+      puts "Sorry but your guess was too low, please try again"
+    else
+      puts "Sh!t, this shouldn't happen!?"  
+    end
+
```
Comment (mathildathompson) [inline students/simon_taylor/secret_number.rb:37]: -This code is clunky with too many if elseif statements in it - try and refactor using a case statement...

File: students/simon_taylor/secret_number.rb [hunk 3]
```diff
@@ -0,0 +1,46 @@
+def secret_number
+  number = random_num
+  #puts "For Testing, Number is #{number}"
+
+  #loop should start here
+
+  exit = false
+
```
Comment (mathildathompson) [inline students/simon_taylor/secret_number.rb:8]: Good attempt at exit!

File: students/simon_taylor/calculator.rb [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    puts "1. Basic Calculator"
+    puts "2. Advanced Calculator"
+    puts "3. Exit"
+    choice = gets.chomp.to_i
+
+    #define labels
+    label1 = "first value"
+    label2 = "second value"
+    label3 = "third value"
+
```
Comment (mathildathompson) [inline students/simon_taylor/calculator.rb:19]: Why are you assigning variables label1, label2, label3 when you go on to re-assign them later?

File: students/simon_taylor/calculator.rb [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+    while choice != 1 && choice != 2 && choice != 3
+      puts "Please select 1, 2 or 3" 
+      choice = gets.chomp.to_i
+    end
+
+    if choice == 3
+      puts "Exiting Calculator..."
+      break
+    end
+
```
Comment (mathildathompson) [inline students/simon_taylor/calculator.rb:30]: Nice use of break here!

File: students/simon_taylor/calculator.rb [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+        result = mortgage_calc(num1,num2,num3)
+        puts "Your loan repayment will be $#{result} per month for #{num3} years"
+    end
+
+    puts "---------------------------------------"
+  
+  end # end of program loop
+
+end
+
```
Comment (mathildathompson) [inline students/simon_taylor/calculator.rb:110]: I like what you are doing here using if statements to filter user input, but the calculator method it too long - it makes the code hard to read. Break it down into smaller methods.

File: students/simon_taylor/calculator.rb (hunk 4 above)
Comment (s-taylor) [inline students/simon_taylor/calculator.rb:19]: The labels are only replaced in some instances, like when using the BMI or Mortgage Calculator and more specific labels are needed i.e. Weight in kg. For the basic functions the labels are unchanged.

**Pull request merged**